sdk/resourcemanager/azure-resourcemanager-test/src/main/java/com/azure/resourcemanager/test/utils/TestDelayProvider.java

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.resourcemanager.test.utils;
+
+import com.azure.core.management.provider.DelayProvider;
+
+import java.time.Duration;
+
+/**
+ * Class helps thread sleep in tests.
+ */
+public class TestDelayProvider implements DelayProvider {
+
+    private final boolean isLiveMode;
+
+    /**
+     * Constructor of TestDelayProvider
+     *
+     * @param isLiveMode the boolean flag for test mode
+     */
+    public TestDelayProvider(boolean isLiveMode) {
+        this.isLiveMode = isLiveMode;
+    }
+
+    @Override
+    public Duration getDelayDuration(Duration delay) {
+        return isLiveMode ? delay : Duration.ofSeconds(1);
```

Comment:
Set deplay in playback as 1. When set to 0, got error for polling during playback.